src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -70,64 +77,80 @@ export function registerFlameChartSearchRoute(router: IRouter<DataRequestHandler
+      const topN = 200; // collect data for the top N unique stack traces
 
       try {
         const esClient = context.core.elasticsearch.client.asCurrentUser;
         const filter = newProjectTimeQuery(projectID!, timeFrom!, timeTo!);
 
-        // const resp = await getCountResponse(context, filter);
+        // Start with counting the results in the index down-sampled by 5^6.
+        // That is in the middle of our down-sampled indexes.
+        const initialExp = 6;
         const resp = await esClient.search({
-          index: 'profiling-events-1000',
+          index: 'profiling-events-5pow' + initialExp,
           body: {
             query: filter,
             size: 0,
             track_total_hits: true,
           },
         });
+        const sampleCountFromInitialExp = resp.body.hits.total.value as number;
 
-        const sampleCountFromSmallestTable = resp.body.hits.total.value as number;
+        console.log('sampleCountFromPow6', sampleCountFromInitialExp);
 
-        const [eventsIndex, sampleRate, estimatedSampleCount] = getSampledTraceEventsIndex(
-          sampleSize,
-          sampleCountFromSmallestTable
+        const eventsIndex = getSampledTraceEventsIndex(
+          targetSampleSize,
+          sampleCountFromInitialExp,
+          initialExp
         );
 
         // eslint-disable-next-line no-console
-        console.log('Index', eventsIndex, sampleRate, estimatedSampleCount);
+        console.log('EventsIndex', eventsIndex);
 
         const resEvents = await esClient.search({
-          index: eventsIndex,
+          index: eventsIndex.name,
           body: {
             size: 0,
-            query: {
-              function_score: {
-                query: filter,
-              },
-            },
+            query: filter,
             aggs: {
-              sample: {
-                sampler: {
-                  shard_size: sampleSize,
+              group_by: {
+                terms: {
+                  field: 'StackTraceID',
+                  size: topN,
                 },
                 aggs: {
-                  group_by: {
-                    terms: {
-                      field: 'StackTraceID',
-                      size: sampleSize,
+                  sum_count: {
+                    sum: {
+                      field: 'Count',
                     },
                   },
                 },
               },
+              total_count: {
+                sum: {
+                  field: 'Count',
+                },
+              },
             },
           },
         });
+        // console.log(JSON.stringify(resEvents, null, 2));
 
         const tracesDocIDs: string[] = [];
-        resEvents.body.aggregations.sample.group_by.buckets.forEach((stackTraceItem: any) => {
+        let sumCount = 0;
+        let docCount = 0;
+        resEvents.body.aggregations.group_by.buckets.forEach((stackTraceItem: any) => {
```

Comment:
If you look into `search_topn.ts` there is a similar function, but with strict typing.
Should we probably try and extract a common helper to "translate" profile-events indices data? (not in this PR)

Reply:
+1 for that